qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/identifier/TomatoId.groovy

```diff
@@ -36,8 +36,17 @@ class TomatoId {
         return new Version(version)
     }
 
+    /**
+     * TODO documentation
+     * @since 1.0.0
+     */
     void increaseVersion() {
         this.version.increaseVersion()
+        //TODO  https://github.com/qbicsoftware/offer-manager-2-portlet/issues/232
+        // query for the latest version first and increase this version
+        // make sure the increased version does not already exist
+        // this should be done somewhere with access to the database
+        // and is part of the business rules
```

Comment:
Why in the identifier? The business logic belongs to the offer, not to the identifier. Can you elaborate maybe your idea?

Reply:
I think that the property that an identifier is unique is a property of the identifier itself. Thus it has to check itself when increasing the version to avoid duplicate versioning. The logic of how to increase belongs here in my opinion.

Reply:
I disagree here. The identifier is just a property of the offer, that is required for the user. The offer itself needs to know, how many versions exist of it. And then create a new version from the latest, if an offer is updated.